backend/fastrtc/websocket.py

```diff
@@ -138,25 +153,25 @@ async def handle_websocket(self, websocket: WebSocket):
                     await self.set_handler(self.stream_id, self)
                 elif message["event"] == "stop":
                     self.quit.set()
-                    self.clean_up(cast(str, self.stream_id))
-                    return
+                    return  # Still runs the `finally` block
                 elif message["event"] == "ping":
                     await websocket.send_json({"event": "pong"})
-
-        except Exception as e:
-            print(e)
-            import traceback
-
-            traceback.print_exc()
-            logger.debug("Error in websocket handler %s", e)
+        except WebSocketDisconnect:
+            # Surprisingly, this leaves `websocket.application_state` as CONNECTED
+            # in the `finally` block, so we use this variable
+            was_disconnected = True
         finally:
             if self._emit_task:
                 self._emit_task.cancel()
             if self._emit_to_queue_task:
                 self._emit_to_queue_task.cancel()
             if self.start_up_task:
                 self.start_up_task.cancel()
-            await websocket.close()
+
+            if not was_disconnected:
+                await websocket.close()
+
+            self.clean_up(cast(str, self.stream_id))
```

Comment:
This wasn't being run if the websocket got disconnected without the "stop" event being sent.

Reply:
Good catch!